### PR TITLE
New commands: New-DbaDbSchema and Get-DbaDbSchema

### LINKED ISSUE #7126: New commands: New-DbaDbSchema and Get-DbaDbSchema
### Summary of new feature

Add new commands to create and get db schemas.

### Proposed technical details (if applicable)

Basic commands that enable creation of schemas and the ability to find schema. Using the Get-DbaDbSchema would allow a caller to do other actions such as altering or dropping the schema.

A follow-on feature request could be to use New-DbaDbSchema from New-DbaDbTable if the schema specified does not exist and the caller has specified a param such as -CreateSchema or -Force.

### PR DESCRIPTION
**Pull request by lancasteradam**
New commands: New-DbaDbSchema and Get-DbaDbSchema

- Utility commands for creating and finding schemas. 

Once these are merged I'd like to explore using the New-DbaDbSchema command in New-DbaDbTable. Perhaps a -CreateSchema param or -Force param can be added to New-DbaDbTable and then it would call New-DbaDbSchema if the schema does not exist.

## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [x] New feature (non-breaking change, adds functionality, fixes #7126> )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [x] Documentation
 - [ ] Build system
 

### Purpose
 
Adding utility commands for creating and finding schemas. The Get-DbaDbSchema can be used to get a schema and then alter the owner or even drop the schema if the objects have been removed (see the command examples).

### Commands to test

See the integration tests.

### CODE REVIEW

File: functions/Get-DbaDbSchema.ps1 [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        PS C:\> $schema.Owner = DBUser2
+        PS C:\> $schema.Alter()
+
+        Finds the TestSchema in the TestDB on the localhost instance and then changes the schema owner to DBUser2
+
+    .EXAMPLE
+        PS C:\> $schema = Get-DbaDbSchema -SqlInstance localhost -Database TestDB -SchemaName TestSchema
+        PS C:\> $schema.Drop()
+
+        Finds the TestSchema in the TestDB on the localhost instance and then drops it. Note: to drop a schema all objects must be transferred to another schema or dropped.
```
Comment (jpomfret) [inline functions/Get-DbaDbSchema.ps1:101]: Can we add an example that demonstrates piping into this function?

File: functions/Get-DbaDbSchema.ps1 [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+        Finds and returns the schemas owned by DBUser1 from the localhost instance.
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbSchema -SqlInstance localhost -Database TestDB -SchemaOwner DBUser1
+
+        Finds and returns the schemas owned by DBUser1 in the TestDB database from the localhost instance.
+
+    .EXAMPLE
+        PS C:\> $schema = Get-DbaDbSchema -SqlInstance localhost -Database TestDB -SchemaName TestSchema
```
Comment (jpomfret) [inline functions/Get-DbaDbSchema.ps1:91]: I think there is room for `Set-DbaDbSchema` and `Remove-DbaDbSchema` too - doesn't have to be in this PR - I've opened #7140 to track this issue.

File: tests/Get-DbaDbSchema.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,134 @@
+$CommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
```
Comment (jpomfret) [inline tests/Get-DbaDbSchema.Tests.ps1:1]: awesome tests! Thanks

File: functions/New-DbaDbSchema.ps1 [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+    .EXAMPLE
+        PS C:\> New-DbaDbSchema -SqlInstance localhost -Database example1 -SchemaName TestSchema1
+
+        Creates the TestSchema1 schema in the example1 database in the localhost instance. The dbo user will be the owner of the schema.
+
+    .EXAMPLE
+        PS C:\> New-DbaDbSchema -SqlInstance localhost -Database example1 -SchemaName TestSchema1, TestSchema2 -SchemaOwner dbatools
+
+        Creates the TestSchema1 and TestSchema2 schemas in the example1 database in the localhost instance and assigns the dbatools as the owner of the schemas.
```
Comment (jpomfret) [inline functions/New-DbaDbSchema.ps1:62]:
```suggestion
        Creates the TestSchema1 and TestSchema2 schemas in the example1 database in the localhost instance and assigns the dbatools user as the owner of the schemas.
```

File: functions/New-DbaDbSchema.ps1 [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+    .EXAMPLE
+        PS C:\> New-DbaDbSchema -SqlInstance localhost, localhost\sql2017 -Database example1 -SchemaName TestSchema1, TestSchema2 -SchemaOwner dbatools
+
+        Creates the TestSchema1 and TestSchema2 schemas in the example1 database in the localhost and localhost\sql2017 instances and assigns the dbatools as the owner of the schemas.
```
Comment (jpomfret) [inline functions/New-DbaDbSchema.ps1:67]:
```suggestion
        Creates the TestSchema1 and TestSchema2 schemas in the example1 database in the localhost and localhost\sql2017 instances and assigns the dbatools user as the owner of the schemas.
```

File: functions/New-DbaDbSchema.ps1 [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+    .EXAMPLE
+        PS C:\> Get-DbaDatabase -SqlInstance localhost, localhost\sql2017 -Database example1 | New-DbaDbSchema -SchemaName TestSchema1, TestSchema2 -SchemaOwner dbatools
+
+        Creates the TestSchema1 and TestSchema2 schemas in the example1 database in the localhost and localhost\sql2017 instances and assigns the dbatools as the owner of the schemas.
```
Comment (jpomfret) [inline functions/New-DbaDbSchema.ps1:72]:
```suggestion
        Creates the TestSchema1 and TestSchema2 schemas in the example1 database in the localhost and localhost\sql2017 instances and assigns the dbatools user as the owner of the schemas.
```

File: functions/New-DbaDbSchema.ps1 [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    #>
```
Comment (jpomfret) [inline functions/New-DbaDbSchema.ps1:73]: Please could you add a piping example?

File: tests/New-DbaDbSchema.Tests.ps1 [hunk 8]
```diff
@@ -0,0 +1,77 @@
+$CommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
```
Comment (jpomfret) [inline tests/New-DbaDbSchema.Tests.ps1:1]: More great tests 👍

**Review by jpomfret — COMMENTED** [review]
Thanks for the great PR - couple of small suggestions - mainly if we could add a piping example to each command?

File: functions/Get-DbaDbSchema.ps1 (hunk 2 above)
Comment (lancasteradam) [reply functions/Get-DbaDbSchema.ps1:91]: Thanks, I agree.

File: functions/Get-DbaDbSchema.ps1 (hunk 1 above)
Comment (lancasteradam) [reply functions/Get-DbaDbSchema.ps1:101]: yes, will do

File: functions/New-DbaDbSchema.ps1 (hunk 7 above)
Comment (lancasteradam) [reply functions/New-DbaDbSchema.ps1:73]: yes, will do

**Comment by lancasteradam** [comment]
Thank you @jpomfret

**Review by jpomfret — APPROVED** [review]

**Comment by jpomfret** [comment]
Thanks for the contribution :)

**Comment by wsmelton** [comment]
@jpomfret you can update psd1 and psm1 files following PRs of this nature so when @potatoqualitee does a release they are exported properly. You can hit me on Discord or Twitter if you need help.

**Comment by jpomfret** [comment]
Hey @lancasteradam - while reviewing the new functions for Set\Remove-DbaDbSchema I realised I didn't notice the parameter name on these is SchemaName.

Would be a breaking change to fix these now but since they are still relatively new I think it would be a good time to fix? Thoughts?

**Comment by lancasteradam** [comment]
@jpomfret, I agree with you. For consistency the param names should match. I'll add a new issue and fix those.

**Pull request merged**